### PR TITLE
-added support for standard array functions

### PR DESCRIPTION
**Pull request by andreiradu**
array.shift, array.unshift, array.sort and array.reverse

sort uses table.sort which has slightly different behavior than js. It works by default with items other than string(which I think is a good thing). It's also not guaranteed to be stable.

### CODE REVIEW

File: src/Transpiler.ts [hunk 1]
```diff
@@ -305,7 +309,11 @@ export abstract class LuaTranspiler {
 
     public transpileLuaLibFunction(func: LuaLibFeature, ...params: string[]): string {
         this.importLuaLibFeature(func);
-        return `__TS__${func}(${params.join(", ")})`;
+        if (params.length > 1) {
+            return `__TS__${func}(${params.join(", ")})`;
+        } else {
+          return `__TS__${func}(${params[0]})`;
```
Comment (Perryvw) [inline src/Transpiler.ts]: What is the purpose of this if/else? `params[0]` and `params.join(", ")` look like they would produce exactly the same result?

File: test/unit/lualib/lualib.spec.ts [hunk 2]
```diff
@@ -320,7 +320,73 @@ export class LuaLibArrayTests {
             Expect(result).toBe(expected[1]);
         }
     }
+    @TestCase("[1, 2, 3]", [3, 2, 1])
+    @TestCase("[1, 2, 3, null]", [3, 2, 1])
+    @TestCase("[1, 2, 3, 4]", [4, 3, 2, 1])
```
Comment (Perryvw) [inline test/unit/lualib/lualib.spec.ts]: Please add edge cases empty array and array with 1 element.

File: test/unit/lualib/lualib.spec.ts [hunk 3]
```diff
@@ -320,7 +320,73 @@ export class LuaLibArrayTests {
...
+        }
+    }
+    @TestCase("[1, 2, 3]", [2, 3])
+    @Test("array.shift")
+    public arrayShift(array: string, expected): void {
+        {
+            // Transpile
+            const lua = util.transpileString(
+                `let testArray = ${array};
+                let val = testArray.shift();
```
Comment (Perryvw) [inline test/unit/lualib/lualib.spec.ts]: I would like to also have a test that checks the value of `val`

File: test/unit/lualib/lualib.spec.ts [hunk 4]
```diff
@@ -320,7 +320,73 @@ export class LuaLibArrayTests {
...
+                return JSONStringify(testArray)`);
+
+            // Execute
+            const result = util.executeLua(lua);
+            // Assert
+            Expect(result).toBe(JSON.stringify(expected));
+        }
+    }
+    @TestCase("[1, 2, 3]", [2, 3])
+    @Test("array.shift")
```
Comment (Perryvw) [inline test/unit/lualib/lualib.spec.ts]: Here also singleton array and empty array would be nice test cases.

File: test/unit/lualib/lualib.spec.ts [hunk 5]
```diff
@@ -320,7 +320,73 @@ export class LuaLibArrayTests {
...
+        }
+    }
+    @TestCase("[3, 4, 5]", [1, 2], [1, 2, 3, 4, 5])
+    @Test("array.unshift")
+    public arrayUnshift(array: string, toUnshift, expected): void {
+        {
+            // Transpile
+            const lua = util.transpileString(
+                `let testArray = ${array};
+                testArray.unshift(${toUnshift});
```
Comment (Perryvw) [inline test/unit/lualib/lualib.spec.ts]: Please also add a test on the return value of unshift.

File: test/unit/lualib/lualib.spec.ts [hunk 6]
```diff
@@ -320,7 +320,73 @@ export class LuaLibArrayTests {
...
+                return JSONStringify(testArray)`);
+
+            // Execute
+            const result = util.executeLua(lua);
+
+            // Assert
+            Expect(result).toBe(JSON.stringify(expected));
+        }
+    }
+    @TestCase("[4, 5, 3, 2, 1]", [1, 2, 3, 4, 5])
```
Comment (Perryvw) [inline test/unit/lualib/lualib.spec.ts]: Same edge cases as above would be nice here.

File: src/Transpiler.ts [hunk 7]
```diff
@@ -305,6 +309,9 @@ export abstract class LuaTranspiler {
 
     public transpileLuaLibFunction(func: LuaLibFeature, ...params: string[]): string {
         this.importLuaLibFeature(func);
+        params = params.filter(element => {
```
Comment (Perryvw) [inline src/Transpiler.ts]: I still don't understand in what situation element.tostring() would be empty string, could you give an example?

Comment (andreiradu) [reply src/Transpiler.ts]: sure, I hit this when calling `array.unshift()` with no params
